### PR TITLE
Migrate to hardhat

### PR DESCRIPTION
**Pull request by HristiyanG**

### CODE REVIEW

**Comment by hswopeams** [comment]
I was able to successfully deploy to Rinkeby using `npx hardhat deploy  --network rinkeby` and verify using `npx hardhat contracts-verify --network rinkeby`, but this only worked after uncommenting the rinkeby network configuration and the INFURA_KEY and DEPLOYER_PRIVATE_KEY constants in hardhat.config.js. Please uncomment these lines. Also, please add some info to the README with instructions on copying .env.example to .env and for using the above-mentioned commads.

**Comment by HristiyanG** [comment]
> I was able to successfully deploy to Rinkeby using `npx hardhat deploy --network rinkeby` and verify using `npx hardhat contracts-verify --network rinkeby`, but this only worked after uncommenting the rinkeby network configuration and the INFURA_KEY and DEPLOYER_PRIVATE_KEY constants in hardhat.config.js. Please uncomment these lines. Also, please add some info to the README with instructions on copying .env.example to .env and for using the above-mentioned commads.

Yep, I agree. Those needs to be uncommented. Locally everything is running as it should but when it comes to the pipeline, it's failing because of those. Mohammed should take care of this. Even if it just creates a dummy `.env` with the required fields.

**Review by hswopeams — CHANGES_REQUESTED** [review]
See changes requested for deployment script and also question regarding Ganache

File: README.md
Comment (hswopeams) [inline README.md:101]: I think here we should mention that if you want to deploy to a testnet like Rinkeby, .env.example needs to be copied to .env and the values provided

File: README.md
Comment (HristiyanG) [reply README.md]: Not really, but it's possible via `hardhat-ganache` plugin, which is described in the README. Updated this line as well.

File: scripts/deploy.js
Comment (HristiyanG) [reply scripts/deploy.js]: Updated the deployment script so that we can deploy prod, dev & test with different setups if required. `env.example` file has been updated as well. In order to have the official BOSON token, it needs to be pre-filled in the `.env`

File: README.md
Comment (HristiyanG) [reply README.md:95]: I am not sure if this is the most appropriate place as this entire section is related to the Local development. From what i see we don't have any procedure steps for a testnet deployment in the README. I could add a few sentences, but this should be on a separate PR IMO

Comment (hswopeams) [reply README.md:95]: OK. Let's revisit after we merge. There are some changes to the README regarding running on Windows that were never merged and probably need to be, so we could do another pass at the README at a later date.

File: scripts/deploy.js
Comment (hswopeams) [reply scripts/deploy.js]: Looks good.

**Review by hswopeams — DISMISSED** [review]

**Comment by hswopeams** [comment]
Approved pending the pipeline being fixed.

**Review by hswopeams — APPROVED** [review]

**Pull request merged**
